src/board.cpp

```diff
@@ -0,0 +1,173 @@
+        std::to_string(sorted_players[i]->get_score()) + " points";
+    GUI::print_item(player_score_text.c_str());
+  }
+  GUI::print_border();
+}
+
+void Board::print_round_result() {
+  int winner_id = check_round_winner(); 
+
+  // If game is over, print final score and exit
+  if (check_game_winner() != CONTINUE_GAME) {
+    print_scores();
+    return;
+  }
+
+  if (winner_id == TIE_GAME) {
+    GUI::print_header("It's a tie!");
+  } else {
+    std::string winner_text = players_[winner_id - 1]->name() + " wins the round!";
+    GUI::print_header(winner_text.c_str());
+  }
+
+  // Sort players by last move, descending
+  std::vector<std::unique_ptr<Player>> sorted_players = players_;
+  std::sort(sorted_players.begin(), sorted_players.end(), move_cmp);
+
+  for (int i = 0; i < num_players(); ++i) {
+    std::string player_move_text = sorted_players[i]->name() + " move: " +
+        std::to_string(sorted_players[i]->get_last_move());
+    GUI::print_item(player_move_text.c_str());
+  }
+
+  // Print the current scores
+  print_scores();
+}
+
+void Board::print_player_moves(unsigned int player_id) {
+  if (player_id > num_players()) {
+    return;
+  }
+  GUI::print_header("NUMBERS ROYALE");
+  std::unique_ptr<Player>& player = players_[player_id - 1];
+  std::string player_move_text = player->name() + " moves:";
+  GUI::print_item(player_move_text.c_str());
+  GUI::print_item(player->get_moves_string().c_str(), true);
+  GUI::print_border();
+}
+
+void Board::add_player(bool is_cpu) {
+  if (is_cpu) {
+    players_.push_back(std::make_unique<CPU>());
+  } else {
+    players_.push_back(std::make_unique<Player>());
+    num_humans_++;
+  }
+}
+
+int Board::check_game_winner() {
+  if (num_players() == 0) {
+    return TIE_GAME;
+  }
+
+  int toWin = board_size_ / num_players();
+  unsigned int max_score = 0;
+  unsigned int winner_id = 0;
+  bool tie = false;
+
+  for (const std::unique_ptr<Player>& player : players_) {
+    // Plurality check
+    if (player->get_score() > toWin) {
+      return player->id();
+    }
+    // Highest score check
+    if (player->get_score() > max_score) {
+      max_score = player->get_score();
+      winner_id = player->id();
+      tie = false;
+    } else if (player->get_score() == max_score) {
+      tie = true;
+    }
+  }
+
+  // If no more moves
+  if (players_[0]->get_moves().empty()) {
+    return tie ? TIE_GAME : winner_id;
+  }
+
+  // Game is not over
+  return CONTINUE_GAME;
+}
+
+int Board::check_round_winner() {
+  int winner_id = 0;
+  int biggest_move = 0;
+  bool tie = false;
+
+  for (int i = 0; i < num_players(); ++i) {
+    int current_move = players_[i]->get_last_move();
+    if (current_move > biggest_move) {
+      biggest_move = current_move;
+      winner_id = players_[i]->id();
+      tie = false;
+    } else if (current_move == biggest_move) {
+      tie = true;
+    }
+  }
+
+  if (tie) {
+    return 0;
+  }
+
```

Comment:
nit
```suggestion
    } else if (current_move == biggest_move) {
      // tie game
      return true;
    }
  }


```

Reply:
While this logic would work for 2 players, this early exit would fail for 3 players or more. You need to check all players.

Reply:
oops my bad, didn't realize it was in a for loop

Reply:
That being said, tie logic would need to be updated for 3 players, so instead of doing that, I may just specify this working for 2 players.

Reply:
correction: The only edge case this doesn't handle is if 2/3 players tie (winning) and the 3rd player loses... Only 1 winner ID is returned and 2 player scores wouldn't be incremented. 

So if I wanted to get rid of of 2+ functionality, this function could be greatly simplified.